src/test/java/org/assertj/core/api/SoftAssertionsTest.java

```diff
@@ -244,7 +244,7 @@ public void should_be_able_to_catch_exceptions_thrown_by_all_proxied_methods() t
       softly.assertThat(new int[] { 24 }).isEqualTo(new int[] { 25 });
 
       softly.assertThat((Iterable<String>) Lists.newArrayList("26")).isEqualTo(Lists.newArrayList("27"));
-      softly.assertThat(Lists.newArrayList("28").iterator()).contains("29");
+      softly.assertThat(Lists.newArrayList("28")).contains("29");
       softly.assertThat(Lists.newArrayList("30")).isEqualTo(Lists.newArrayList("31"));
```

Comment:
add an Iterator assertion (hasNext will do) to make sure they work with soft assertions (easier to add it just before the catch block to avoid changing the error messages order afterwards.